packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts

```diff
@@ -137,10 +140,12 @@ export async function updateamplifyMetaAfterPush(resources) {
     // Skip hash calculation for imported resources
     if (resources[i].serviceType !== 'imported') {
       const sourceDir = path.normalize(path.join(pathManager.getBackendDirPath(), resources[i].category, resources[i].resourceName));
-      const hashDir = await getHashForResourceDir(sourceDir);
-
-      amplifyMeta[resources[i].category][resources[i].resourceName].lastPushDirHash = hashDir;
-      amplifyMeta[resources[i].category][resources[i].resourceName].lastPushTimeStamp = currentTimestamp;
+      // skip hashing deleted resources
+      if (fs.pathExistsSync(sourceDir)) {
+        const hashDir = await getHashForResourceDir(sourceDir);
```

Comment:
Without calculating the hash - did you check if the resource status still shows as "Deleted"?

Reply:
after pushing, the deleted resource doesn't show up at all (which is expected)
Also checked delete -> push -> new folder -> pull -> status and the deleted resource doesn't show up